articles/documenting-node.js-rest-api-using-swagger/index.md

```diff
@@ -0,0 +1,550 @@
+            example: false,
+          },
+        },
+      },
+      Error: {
+        type: "object",
+        properties: {
+          message: {
+            type: "string",
+          },
+          internal_code: {
+            type: "string",
+          },
+        },
+      },
+    },
+  },
+};
+```
+
+Since our API is not that complex, we have only included the `schemas` in the components.
+
+In the `schemas`, we have defined the `id`, `Todo`, `TodoInput`, and `Error`. For each schema definition, we are describing its respective data using keys such as `type`, `description`, `example`, and `properties`.
+
+To test this:
+
+- Ensure that the development server is running.
+
+- In your browser, refresh the documentation page.
+
+- In case of errors, revisit the steps. Else, the following section should be added to the documentation page:
+
+![schemas_screenshot](/documenting-node.js-rest-api-using-swagger/schemas-screenshot.png)
+
+### Documenting API's paths
+
+Paths are the routes that are to be accessed. Each route is different from the other based on the method of operation and the data passed.
+
+To document the paths in our API, we will cover each route separately.
+
+#### Getting todos i.e /todos route
+
+When getting the todos, we are sending a `GET` request to `/todos`.
+
+To document this path, we edit the `/docs/todos/get-todos.js` file as follows:
+
+```javascript
+module.exports = {
+  get: {
+    tags: ["Todo CRUD operations"],
+    description: "Get todos",
+    operationId: "getTodos",
+    parameters: [],
+    responses: {
+      200: {
+        description: "Todos were obtained",
+        content: {
+          "application/json": {
+            schema: {
+              $ref: "#/components/schemas/Todo",
+            },
+          },
+        },
+      },
+    },
+  },
+};
+```
+
+From above, we have:
+
+- Specified the method of the operation, `get`.
+- Specified the tags of the operation.
+- Described the operation with a short `description` and an `operationId`.
+- Defined the `parameters`. For this case, there is none.
+- Defined the expected `responses`. Only a `200` response is expected since we are fetching data. In the response, we also describe the content type and match the schema to that from the components using `$ref`.
+
+To test this:
+
+- Ensure that the development server is running.
+- In your browser, refresh the documentation page.
+- In case of errors, revisit the steps. Else the following section should be added to the page:
+
+![get_todos_screenshot](/documenting-node.js-rest-api-using-swagger/get-todos-screenshot.png)
+
+- To test it out, click on the `Try it out`, then click `Execute`, and then examine the server response.
+
+#### Getting single todo (/todos/:id)
+
+When getting a single todo, we are sending a `GET` request to `/todos/:id`. The `:id` is for dynamic `id`.
+
+To document this path, we will edit the `/docs/todos/get-todo.js` file as follows:
+
+```javascript
+module.exports = {
```

Comment:
Add comments if possible. It would improve readability

Reply:
done.